bin/alpen-client/src/gossip.rs

```diff
@@ -0,0 +1,218 @@
+//! Gossip event handling task for managing peer connections and broadcasting blocks.
+
+use std::collections::{HashMap, HashSet};
+
+use alloy_primitives::B256;
+use alpen_reth_node::{AlpenGossipCommand, AlpenGossipEvent, AlpenGossipMessage};
+use reth_network_api::PeerId;
+use reth_primitives::Header;
+use reth_provider::CanonStateNotification;
+use strata_acct_types::Hash;
+use strata_primitives::buf::Buf32;
+use tokio::{
+    select,
+    sync::{broadcast, mpsc},
+};
+use tracing::{debug, error, info, warn};
+
+/// Configuration for the gossip task.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub(crate) struct GossipConfig {
+    /// Sequencer's public key for signature validation.
+    pub sequencer_pubkey: Buf32,
+
+    /// Sequencer's private key for signing (only in sequencer mode).
+    #[cfg(feature = "sequencer")]
+    pub sequencer_privkey: Buf32,
+
+    /// Maximum number of seen packages to track before clearing the set.
+    pub max_seen_packages: usize,
+}
+
+/// Creates the gossip event handling task.
+///
+/// This task manages:
+///
+/// - Connection tracking (establish/close)
+/// - Receiving gossip messages and forwarding block hashes to engine control
+/// - Broadcasting new canonical blocks to connected peers
+pub(crate) async fn create_gossip_task(
```

Comment:
nit: This is a very big function. It might be worthwhile to separate out the `select!` handlers.

Reply:
Yes see bc639675794ece87af9b1ca2aad7f5e81861f8da